Source/Diagnostics/ReducedDiags/FieldReduction.H

```diff
@@ -0,0 +1,184 @@
+            {
+                 // 0.5 is here because position are computed on the cell centers
+                const amrex::Real x = (i + 0.5_rt)*dx[0] + real_box.lo(0);
+#if (AMREX_SPACEDIM==2)
+                const amrex::Real y = 0._rt;
+                const amrex::Real z = (j + 0.5_rt)*dx[1] + real_box.lo(1);
+#else
+                const amrex::Real y = (j + 0.5_rt)*dx[1] + real_box.lo(1);
+                const amrex::Real z = (k + 0.5_rt)*dx[2] + real_box.lo(2);
+#endif
+                const amrex::Real Ex_interp = CoarsenIO::Interp(arrEx, Extype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                const amrex::Real Ey_interp = CoarsenIO::Interp(arrEy, Eytype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                const amrex::Real Ez_interp = CoarsenIO::Interp(arrEz, Eztype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                const amrex::Real Bx_interp = CoarsenIO::Interp(arrBx, Bxtype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                const amrex::Real By_interp = CoarsenIO::Interp(arrBy, Bytype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                const amrex::Real Bz_interp = CoarsenIO::Interp(arrBz, Bztype, cellCenteredtype,
+                                        reduction_coarsening_ratio, i, j, k, reduction_comp);
+                return reduction_function_parser(x, y, z, Ex_interp, Ey_interp, Ez_interp,
+                                                          Bx_interp, By_interp, Bz_interp);
+            });
+        }
+
+        amrex::Real reduce_value = amrex::get<0>(reduce_data.value());
+
+        // MPI reduce
+        if (std::is_same<T, amrex::ReduceOpMax>::value)
+        {
+            amrex::ParallelDescriptor::ReduceRealMax(reduce_value);
+        }
+        if (std::is_same<T, amrex::ReduceOpMin>::value)
+        {
+            amrex::ParallelDescriptor::ReduceRealMin(reduce_value);
+        }
+        if (std::is_same<T, amrex::ReduceOpSum>::value)
+        {
+            amrex::ParallelDescriptor::ReduceRealSum(reduce_value);
+        // If reduction operation is a sum, multiply the value by the cell volume so that the
+        // result is an integration over the simulation domain rather than simply a sum of the
+        // values on the grid points.
+#if (AMREX_SPACEDIM==2)
+            reduce_value *= dx[0]*dx[1];
+#else
+            reduce_value *= dx[0]*dx[1]*dx[2];
+#endif
+        }
+
+        // Fill output array
+        m_data[0] = reduce_value;
+
+        // m_data now contains an up to date value of the reduced field quantity
+    }
+
+};
+
+#endif // WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_
```

Comment:
```suggestion
#endif // WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_

```
I think that we typically add a newline at the end of an `.H` file (although this is no longer necessary since C++11)

Reply:
I think there's already a newline at the end of the file. (it doesn't show up on github, but I see it on my IDE)